### PR TITLE
[POC] Add contexts for time bounded tests

### PR DESCRIPTION
**Pull request by skshetry**
## Description

This implements a proof of concept to do time bounded tests.
@phil-davis 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/acceptance/features/bootstrap/TimeContext.php [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+
+	/**
+	 * @AfterScenario @time-bound
+	 *
+	 * @return void
+	 */
+	public function revertSystemtime() {
+		for ($i = 0; $i < $this->weeksPassed; $i++) {
+			echo("Reverting back a week\n");
+			\shell_exec("date --set 'last week'");
```
Comment (phil-davis) [inline tests/acceptance/features/bootstrap/TimeContext.php]: The challenge will be to see which environments (local developers, drone, Travis and other systems-under-test) will allow the web-server to execute the ``date --set``  command.

**Review by phil-davis — COMMENTED** [review]
I found https://github.com/wolfcw/libfaketime
Maybe this, or something similar, can fake the time underneath Apache/PHP without even setting the real system time.

**Comment by skshetry** [comment]
@phil-davis I also found this: [hnw/php-timecop](https://github.com/hnw/php-timecop).

**Comment by skshetry** [comment]
@phil-davis I also found this: (hnw/php-timecop)[https://github.com/hnw/php-timecop]

**Pull request closed without merging**